analyzers/src/SonarAnalyzer.CSharp/Rules/MessageTemplates/MessageTemplateAnalyzer.cs

```diff
@@ -54,7 +55,7 @@ public sealed class MessageTemplateAnalyzer : SonarDiagnosticAnalyzer
     private static ArgumentSyntax TemplateArgument(InvocationExpressionSyntax invocation, SemanticModel model) =>
         TemplateArgument(invocation, model, KnownType.Microsoft_Extensions_Logging_LoggerExtensions, MicrosoftExtensionsLogging, "message")
         ?? TemplateArgument(invocation, model, KnownType.Serilog_Log, Serilog, "messageTemplate")
-        ?? TemplateArgument(invocation, model, KnownType.Serilog_ILogger, Serilog, "messageTemplate")
+        ?? TemplateArgument(invocation, model, KnownType.Serilog_ILogger, Serilog, "messageTemplate", checkDerivedTypes: true)
```

Comment:
I did not add a test for this because Serilog's Logger requires a lot of methods to implement and I do not think it is worth it.

Reply:
It's enough to check it on a single method (e.g. Warning) and assume it works on all the others.

Reply:
The problem is that I need to implement `ILogger` with a custom class, and the implementation will be like 300 lines of `throw NotImplemented`, just to cover this one bool flag.

Reply:
Use the [NullLogger](https://nlog-project.org/documentation/v5.0.0/html/T_NLog_NullLogger.htm) class.

Reply:
That's part of NLog, not Serilog.
Or am I missing something?

Reply:
My bad. How about `Serilog.Core.Logger`, which implements `ILogger`?

Reply:
Good find, I missed that.
It's only compatible with .NET (hence the Core in the path), so I made a stand-alone test just for this.
Thanks!